osu.Game/Graphics/UserInterface/OsuTextBox.cs

```diff
@@ -100,6 +100,12 @@ public OsuCaret()
 
             public Color4 SelectionColour { get; set; }
 
+            public Color4 CaretColour
+            {
+                get => beatSync.Colour;
+                set => beatSync.Colour = value;
+            }
```

Comment:
This would cause blending as you're changing the colour of an internal child of the caret while colour transformations below are applied to the caret, what you really want to do is something like this:
```csharp
private Color4 caretColour;

public Color4 CaretColour
{
    get => caretColour;
    set
    {
        if (caretColour == value)
            return;

        caretColour = value;
        if (!HasSelection) // move HasSelection to here.
            this.FadeColour(value, 200, Easing.Out);
    }
}

...
// in DisplayAt(position, null):
this.FadeColour(CaretColour, 200, Easing.Out); // (instead of Color4.White)
```
same should be applied for `SelectionColour` with the exception of fading colour if there's a selection.